src/codegen.cpp

```diff
@@ -1298,7 +1298,7 @@ void jl_extern_c(jl_function_t *f, jl_value_t *rt, jl_value_t *argt, char *name)
             assert(!imaging_mode);
 #if defined(USE_MCJIT) || defined(USE_ORCJIT)
             jl_ExecutionEngine->addGlobalMapping(llvmf->getName(),
-                    (void*)getAddressForOrCompileFunction(llvmf));
```

Comment:
Seems that the `JuliaOJIT` and `llvm::ExecutionEngine` doesn't agree on this API. I guess we should probably fix `JuliaOJIT::addGlobalMapping`.